packages/ember-metal/lib/property_events.js

```diff
@@ -116,22 +116,23 @@ function notifyPropertyChange(obj, keyName, _meta) {
   }
 }
 
-let DID_SEEN = null;
+let SEEN_MAP = new Map();
+let IS_TOP_SEEN_MAP = true;
 
 // called whenever a property has just changed to update dependent keys
 function dependentKeysDidChange(obj, depKey, meta) {
   if (meta.isSourceDestroying() || !meta.hasDeps(depKey)) { return; }
-  let seen = DID_SEEN;
-  let top = seen === null;
+  let seen = SEEN_MAP;
 
-  if (top) {
-    seen = DID_SEEN = new Map();
+  if (IS_TOP_SEEN_MAP) {
+    IS_TOP_SEEN_MAP = false;
   }
 
   iterDeps(notifyPropertyChange, obj, depKey, seen, meta);
 
-  if (top) {
-    DID_SEEN = null;
+  if (IS_TOP_SEEN_MAP) {
```

Comment:
Won't this always be false? did you mean to set the local `top = IS_TOP_SEEN_MAP` and use top in the conditionals?

Do we have a test that covers this?

Reply:
thanks for catch, fixed, not sure how to test this, though it is doing same thing as before